packages/SwingSet/src/vats/comms/state.js

```diff
@@ -28,17 +29,31 @@ export function makeState(identifierBase = 0) {
     remotes: new Map(), // remoteNN -> { remoteID, name, fromRemote/toRemote, etc }
     names: new Map(), // name -> remoteNN
 
+    fromKernel: new Map(), // o+NN/o-NN/p+NN/p-NN -> loNN/lpNN
+    toKernel: new Map(), // loNN/lpNN -> o+NN/o-NN/p+NN/p-NN
+
+    // o+NN/o-NN comms management meta-objects that shouldn't be in any clist
```

Comment:
I believe the receiver is an import.

Reply:
I believe the transmitter is an import.

Reply:
Ah, ok. The controller and the receivers are legitimate targets for a `dispatch.deliver`, the transmitters are not (the comms vat sends messages *to* the transmitter object, but of course can never see that ID in the target to a message, because the transmitter lives on a different vat entirely). However all three are not supposed to show up in the arguments of a non-meta-message, so we can use `metaObjects` to enforce that prohibition across all three categories.

Technically it wouldn't cause any problems if a transmitter were cited in a normal message.. it'd be foolish security fault for a closely-held transmitter to be exposed that way, but it's just a normal reference to something in a different vat, so remote calls that target or reference them would be delivered normally. The main reason for prohibiting the controller and receiver objects from being exposed to cross-machine messages is that a remote message that targets one of them would *not* work: the comms vat would receive `deliverFromRemote` and be expected to deliver it, but the kernel-facing clist only cites vat *imports*, not exports, so `sendToKernel` would have no way to reach the meta-object. And we certainly don't want to add a fourth c-list `getMetaForLocal`.

But yeah, prohibiting transmitters is fine, and might save someone from a mistake down the line, and in that case this holds both `o+NN` and `o-NN` identifiers
.